new/components/basic/form/BasicForm.tsx

```diff
@@ -1,25 +1,30 @@
 import React, { BaseSyntheticEvent, FormHTMLAttributes, PropsWithChildren } from "react";
 import { View } from "react-native";
 import BasicButton from "@/components/basic/buttons/BasicButton";
+import { loopValidations } from "@/lib/validations";
 
 interface FormProps extends PropsWithChildren<FormHTMLAttributes<HTMLFormElement>> {
-    onSubmitCallback?: (e: BaseSyntheticEvent, formData: FormData) => void
+    onSubmitCallback?: (e: BaseSyntheticEvent, formData: FormData) => void,
 }
 
 export default function BasicForm({ onSubmitCallback, children, ...props }: FormProps) {
```

Comment:
suggest to useContext and wrap the whole from with Provider. It will be a lot easier to capture the validation result as well as input value